src/app/BUILD.gn

```diff
@@ -294,13 +297,23 @@ static_library("interaction-model") {
       "clusters/ota-provider/ota-provider.cpp",
       "dynamic_server/AccessControl.cpp",
       "dynamic_server/AccessControl.h",
+    ]
+
+    # DynamicDispatcher is actually an ember-override that takes over
+    # util/attribute-storage.cpp and util/attribute-table.cpp functions.
+    #
+    # We likely should formalize and change this with a proper DataModel::Provider that
```

Comment:
DynamicDispatcher should be switched to use a DataModel::Provider, indeed.  The problem is right now it has no way to provide one, right?

Need tracking issues for:

1. A way to inject a data model provider during stack bringup instead of defaulting in the codegen one.  For DynamicDispatcher, having this just on the controller factory would be enough.
2. Making DynamicDispatcher actually use a provider.

Both should be linked from here, presumably.

Reply:
I cannot do this in one PR: existing code assumes embed is built in and everything evolves around that. My plan would be that every app would have to set a provider before continuing and the default provider (if any) should just be a noop/deny everything.

Before doing that though we have to change every application to set a provider and for controller sides we have to figure out where such provider is set (probably in some sort of platform init call).

Reply:
Created #36230 for a generic coverage of things. I will split it up when starting to work on it - for now just keeping the list of checkmarks of todo items.